addons/controllers/packageinstallstatus_controller.go

```diff
@@ -0,0 +1,448 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/controllers/remote"
+	clusterapiutil "sigs.k8s.io/cluster-api/util"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	kapppkgiv1alpha1 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/packaging/v1alpha1"
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/constants"
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/types"
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/util"
+	"github.com/vmware-tanzu/tanzu-framework/addons/predicates"
+	runtanzuv1alpha3 "github.com/vmware-tanzu/tanzu-framework/apis/run/v1alpha3"
+	tkrconstants "github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkr/pkg/constants"
+)
+
+const kappCtrlPkgPrefix = "kapp-controller"
+
+type ClusterRole int
+
+const (
+	clusterRoleManagement ClusterRole = iota
+	clusterRoleWorkload
+)
+
+// PackageInstallStatusReconciler contains the reconciler information for PackageInstallStatus controller
+type PackageInstallStatusReconciler struct {
+	Ctx        context.Context
+	Log        logr.Logger
```

Comment:
I would advice using the context provided by controller runtime when calling Reconcile and then getting the logger from there

Reply:
I have another comment about it above. I think this should be discussed too, but given that it is used across controllers in this repo I would keep that discussion separate from this PR.

Reply:
@fabriziopandini So you mean passing along the following context, am I right? ```ctx := ctrl.SetupSignalHandler()```

Reply:
Let's continue the discussion in: https://github.com/vmware-tanzu/tanzu-framework/pull/1781#discussion_r833981847

Fabrizio meant using the context passed into the Reconcile func by controller-runtime.
I went into more details in the discussion above, but tl;dr I would keep this new controller consistent with other controllers in tanzu-framework and consider a change in a follow-up.

Inconsistency or refactoring all of them in this PR are both bad options :)

Reply:
Thank you. I reverted the changes in my previous commit to get logger from context for consistency with the rest of the code